docs/types/operators.rst

```diff
@@ -26,6 +26,19 @@ except for comparison operators where the result is always ``bool``.
 The operators ``**`` (exponentiation), ``<<``  and ``>>`` use the type of the
 left operand for the operation and the result.
 
+Ternary Operator
+----------------
+The result of the ternary operator (``... ? ... : ...``) is never a number literal, even if all of its operands are number literals.
```

Comment:
Now that this is its own section, maybe we should start with a more generic introduction unrelated to number literals.